src/coap_tcp.c

```diff
@@ -96,7 +92,10 @@ coap_socket_connect_tcp1(coap_socket_t *sock,
       coap_log(LOG_WARNING,
                "coap_socket_connect_tcp1: setsockopt SO_REUSEADDR: %s\n",
                coap_socket_strerror());
-    if (bind(sock->fd, &local_if->addr.sa, local_if->size) == COAP_SOCKET_ERROR) {
+    if (bind(sock->fd, &local_if->addr.sa,
+             local_if->addr.sa.sa_family == AF_INET ?
```

Comment:
This should not be necessary. If `local_if` is properly set, `local_if.size` would be `sizeof(struct sockaddr_in)` for family AF_INET. That is the very meaning of the size field.

Reply:
This addresses issue fixed  #396 which was fixed in coap_io.c and this code is now a part of coap_tcp.c

Reply:
True, but not less broken as it was at [that time](https://github.com/obgm/libcoap/issues/150#issuecomment-527947447). But okay, as this was the workaround at that time, I am okay with it for now. I will look into a proper fix another time.